rhombus/private/map.rkt

```diff
@@ -179,8 +181,15 @@
        (cond
          [(and (pair? arg) (pair? (cdr arg)) (null? (cddr arg)))
           (loop (hash-set ht (car arg) (cadr arg)) (cdr args))]
+         [(and (treelist? arg) (= 2 (treelist-length arg)))
+          (loop (hash-set ht (treelist-ref arg 0) (treelist-ref arg 1)) (cdr args))]
+         [(and (listable? arg) (let ([arg (to-list #f arg)])
+                                 (and (pair? arg) (pair? (cdr arg)) (null? (cddr arg))
+                                      arg)))
+          => (lambda (arg)
+               (loop (hash-set ht (car arg) (cadr arg)) (cdr args)))]
          [else
-          (raise-argument-error* who rhombus-realm "[_, _]" arg)])])))
+          (raise-argument-error* who rhombus-realm "Listable[_, _]" arg)])])))
```

Comment:
This annotation is very wrong; it should be something like `Listable.to_list && matching([_, _])` (again, we should probably add a `Listable.to_list` annotation for this purpose).

Reply:
HF